src/core/stream_send.c

```diff
@@ -1096,7 +1096,8 @@ QuicStreamSendWrite(
         uint16_t StreamFrameLength = AvailableBufferLength - Builder->DatagramLength;
         QuicStreamWriteStreamFrames(
             Stream,
-            Builder->PacketType == QUIC_INITIAL,
+            ((!QuicConnIsVersion2(Stream->Connection) && Builder->PacketType == QUIC_INITIAL_V1) ||
+            (QuicConnIsVersion2(Stream->Connection) && Builder->PacketType == QUIC_INITIAL_V2)),
```

Comment:
Let's put this in a separate (const) variable above. As we add more versions, the logic will likely grow and I don't want to cram it all into here.

Reply:
Should it be a flag on the Builder?

Reply:
Maybe...

Reply:
let's see if the current changes improve the perf tests, and consider adding to the Building as a performance improvement.